cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -248,38 +236,95 @@ where
     }
 }
 
+pub fn get_derivation_info(
+    path: &str,
+) -> Result<serde_json::Map<String, serde_json::Value>, PublishError> {
+    let mut info_cmd = Command::new(&*NIX_BIN);
+    info_cmd.arg("derivation").arg("show").arg(path);
+    let output = info_cmd
+        .output()
+        .map_err(|e| PublishError::MissingDerviationInfo(e.to_string()))?;
+
+    let stdout = String::from_utf8_lossy(&output.stdout);
+    let json: serde_json::Value = serde_json::from_str(&stdout)
+        .map_err(|e| PublishError::MissingDerviationInfo(e.to_string()))?;
+    Ok(json.as_object().unwrap().clone())
+}
+
+pub fn check_build_metadata_from_storepath(
+    pkg: &str,
+    storepath: &str,
+) -> Result<CheckedBuildMetadata, PublishError> {
+    let drv_json = get_derivation_info(storepath)?;
+
+    // Error out if there was no derivation info found
+    if drv_json.is_empty() || drv_json.keys().next().is_none() {
+        return Err(PublishError::MissingDerviationInfo(
+            "No derivation info found".to_string(),
+        ));
+    }
+    let drv_path = drv_json.keys().next().unwrap().to_string();
+    let name = drv_json[&drv_path]["name"].as_str().unwrap().to_string();
+    let system_str = drv_json[&drv_path]["system"].as_str().unwrap();
+    // TODO - There is a drv_json[drv_path]["env"]["version"] in some derivations
+    // that looks like the source of the version that is then later included in
+    // the 'name' Once we include 'version' in the builders, we should look into
+    // grabbing it from there.  Perhaps there won't ever be a 'version' where we
+    // are currently looking for it?  Same for 'description'?
+    let version = drv_json[&drv_path]["version"]
+        .as_str()
+        .map(|s| s.to_string());
+    let description = drv_json[&drv_path]["description"]
+        .as_str()
+        .map(|s| s.to_string());
+
+    let system = SystemEnum::from_str(system_str).map_err(|e| {
+        PublishError::UnsupportedEnvironmentState(format!("Unable to identify system: {e}"))
+    })?;
+
+    let outputs = catalog_api_v1::types::Outputs(
+        drv_json[&drv_path]["outputs"]
+            .as_object()
+            .unwrap()
+            .iter()
+            .map(|(name, value)| Output {
+                name: name.to_string(),
+                store_path: value["path"].as_str().unwrap().to_string(),
+            })
+            .collect(),
```

Comment:
Are we movig description into the derivation explicitly?
usually description would eb a meta attribute in nix which explicitly does _not_ influence the drv.
This also causes a divide between manifest builds and the upcoming NEF.
Besides, manifest builds still exist in "evaluatable form" at some point,
not making use of that looks like a missed opportunity.

Reply:
in reference to the todo: if we do have a split, i think reading this metadata from the manifest/lockfile is a better source of truth than the drv.

Reply:
Getting description and version from the manifest as discussed.  I think that addresses this comment.